src/components/FlawComments.vue

```diff
@@ -1,149 +1,240 @@
 <script setup lang="ts">
-import { computed, ref } from 'vue';
+import { computed, onMounted, ref, watch } from 'vue';
 import LabelTextarea from '@/components/widgets/LabelTextarea.vue';
 import sanitizeHtml from 'sanitize-html';
 import { osimRuntime } from '@/stores/osimRuntime';
 import { useUserStore } from '@/stores/UserStore';
 import { DateTime } from 'luxon';
+import Tabs from '@/components/widgets/Tabs.vue';
+import { useInternalComments } from '@/composables/useInternalComments';
+import { taskUrl } from '@/services/JiraService';
 
 const userStore = useUserStore();
 
 const props = defineProps<{
   comments: any[];
+  taskKey: string | null | undefined;
   isSaving: boolean;
 }>();
-const newPublicComment = ref('');
+
+enum CommentTab {
+    Public,
+    Internal,
+    System,
+}
+
+const {
+  internalComments,
+  addInternalComment,
+  loadInternalComments,
+  isSavingInternalComment,
+  isLoadingInternalComments,
+  internalCommentsAvailable,
+} = useInternalComments(props.taskKey ?? '');
+
+watch(isSavingInternalComment, () => {
+  emit('disableForm', isSavingInternalComment.value);
+});
+
+const newComment = ref('');
 const isAddingNewComment = ref(false);
 
 const emit = defineEmits<{
   'comment:addPublicComment': [value: any, value: any];
+  'disableForm': [value: boolean];
 }>();
 
 const SYSTEM_EMAIL = 'bugzilla@redhat.com';
 
+const tabLabels = ['Public Comments', 'Internal Comments', 'System Comments'];
+const selectedTab = ref(CommentTab.Public);
+const handleTabChange = (index: number) => {
+  selectedTab.value = index;
+};
 
-function handleClick() {
-  emit('comment:addPublicComment', newPublicComment.value, userStore.userName);
-  isAddingNewComment.value = false;
-}
 
 type CommentFilter = 'public' | 'private' | 'system';
-type CommentActiveFilters = Record<CommentFilter, boolean>;
 type CommentFilterFunctions = Record<CommentFilter, (comment: any) => boolean>;
 
-const selectedFilters = ref<CommentActiveFilters>({
-  public: true,
-  private: false,
-  system: false,
-});
-
 const filterFunctions: CommentFilterFunctions = {
   public: (comment: any) => !comment.is_private,
   private: (comment: any) => comment.is_private,
   system: (comment: any) => comment.creator === SYSTEM_EMAIL,
 };
 
-const activeFilters = computed(() => {
-  if (Object.values(selectedFilters.value).every((isActive) => !isActive)) {
-    return () => true;
-  }
-
-  const activeFilterFunctions = Object.entries(selectedFilters.value)
-    .filter(([, isActive]) => isActive)
-    .map(([filter]) => filterFunctions[filter as CommentFilter]);
+function parseOsidbComments(comments: any) {
```

Comment:
We have the comment schema defined as `ZodFlawCommentSchema`, 
maybe we could add the type and use it in this component ? (not only this function, but on the props as well)
```ts
export type ZodFlawCommentSchemaType = z.infer<typeof ZodFlawCommentSchema>
```
If the schema changes, these types will be valid and will break at runtime. 

> Personally I don't like using `any` but if it's fine for the team it's fine for me